core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -341,209 +370,88 @@ private void syncRemoteToLocal() {
+        latestDocumentMap.put(latestDocument.get("_id"), latestDocument);
+      }
+
+      for (final Map.Entry<BsonValue, ChangeEvent<BsonDocument>> eventEntry :
+          remoteChangeEvents.entrySet()) {
+        logger.info(String.format(
+            Locale.US,
+            "t='%d': syncRemoteToLocal consuming event of type: %s",
+            logicalT,
+            eventEntry.getValue().getOperationType()));
 
-      for (final Map.Entry<BsonValue, ChangeEvent<BsonDocument>> eventEntry : remoteChangeEvents) {
         final CoreDocumentSynchronizationConfig docConfig =
-            nsConfig.getSynchronizedDocument(eventEntry.getKey());
+            nsConfig.getSynchronizedDocument(eventEntry.getKey().asDocument().get("_id"));
+
         if (docConfig == null) {
           // Not interested in this event.
           continue;
         }
+
         unseenIds.remove(docConfig.getDocumentId());
+        latestDocumentIds.remove(docConfig.getDocumentId());
         syncRemoteChangeEventToLocal(nsConfig, docConfig, eventEntry.getValue(), localColl);
       }
 
       for (final BsonValue docId : unseenIds) {
         final CoreDocumentSynchronizationConfig docConfig =
             nsConfig.getSynchronizedDocument(docId);
         if (docConfig == null) {
+          // means we aren't actually synchronizing on this remote doc
           continue;
         }
 
-        // We need to look up the document if:
-        // 1. We've never recorded a version and we have no record of the document.
-        if (docConfig.getLastKnownRemoteVersion() == null
-            && localColl.find(getDocumentIdFilter(docConfig.getDocumentId())).first() == null) {
-          logger.info(String.format(
-              Locale.US,
-              "t='%d': syncRemoteToLocal ns=%s documentId=%s no change stream info; looking up "
-                  + "the document",
-              logicalT,
-              nsConfig.getNamespace(),
-              docConfig.getDocumentId()));
-          final CoreRemoteMongoCollection<BsonDocument> remoteColl =
-              getRemoteCollection(nsConfig.getNamespace());
-          final ChangeEvent<BsonDocument> remoteChangeEvent =
-              getSynthesizedRemoteChangeEventForDocument(remoteColl, docConfig.getDocumentId());
-          syncRemoteChangeEventToLocal(nsConfig, docConfig, remoteChangeEvent, localColl);
-        }
-      }
-    }
-
-    logger.info(String.format(
-        Locale.US,
-        "t='%d': syncRemoteToLocal END",
-        logicalT));
-  }
-
-  /**
-   * Attempts to synchronize the given remote change event into the local database.
-   *
-   * @param nsConfig          the namespace configuration.
-   * @param docConfig         the document configuration related to the event.
-   * @param remoteChangeEvent the remote change event to synchronize into the local database.
-   * @param localColl         the local collection where the document lives.
-   */
-  private void syncRemoteChangeEventToLocal(
```

Comment:
Reminder to move this to where it was to make the diff easier. I can read it now but it will be better for future reference.